src/uart.rs

```diff
@@ -1899,15 +1899,15 @@ fn new_common<UART: Uart>(
         uart_driver_install(
             UART::port(),
             if rx.is_some() {
-                config.rx_fifo_size as _
+                config.rx_fifo_size
             } else {
-                0
-            },
+                UART_FIFO_SIZE * 2
+            } as _,
             if tx.is_some() {
-                config.tx_fifo_size as _
+                config.tx_fifo_size
             } else {
-                0
-            },
+                UART_FIFO_SIZE * 2
```

Comment:
However.... why do we need this? Are you having problems with TX as well, or are you doing it "just in case"? IMO this is not necessary, because the TX assert is slightly different: https://github.com/espressif/esp-idf/blob/a322e6bdad4b6675d4597fb2722eea2851ba88cb/components/driver/uart/uart.c#L1589

Reply:
Agree. Cancel changes